pkg/api/state.go

```diff
@@ -173,37 +173,43 @@ func (s *service) generateState(ctx context.Context, pid int32) (*task.ProcessSt
 	// if it doesn't - we inheritFd it?
 	cwd, err := p.Cwd()
 	if err != nil {
-		return nil, nil
+		return state, nil
 	}
 
 	// system information
-	cpuinfo, _ := cpu.Info()
-	vcpus, _ := cpu.Counts(true)
-	state.CPUInfo = &task.CPUInfo{
-		Count:      int32(vcpus),
-		CPU:        cpuinfo[0].CPU,
-		VendorID:   cpuinfo[0].VendorID,
-		Family:     cpuinfo[0].Family,
-		PhysicalID: cpuinfo[0].PhysicalID,
+	cpuinfo, err := cpu.Info()
+	vcpus, err := cpu.Counts(true)
+	if err == nil {
```

Comment:
I think you can avoid this + a lot of err == nil and err checking in the function if you declare `var state task.ProcessState` at the top of the func, then instead of returning nil it'll return an empty struct? I don't remember why this was set to be a pointer in the first place

Reply:
Great idea.

Reply:
Oh, turns out we need to return pointe because that struct contains a lock. It's a proto object.